remote_execution/oss/re_grpc/src/client.rs

```diff
@@ -258,120 +210,126 @@ impl REClient {
                 action_digest,
                 rpc_status.code,
                 rpc_status.message
             )),
             OpResult::Response(any) => {
-                let execute_response: GExecuteResponse = GExecuteResponse::decode(&any.value[..])?;
-                // note: the execute_response.status field is undefined when response is successful
-                let action_result = execute_response
+                let execute_response_grpc: GExecuteResponse =
+                    GExecuteResponse::decode(&any.value[..])?;
+                // note: the execute_response_grpc.status field is undefined when response is successful
+                let action_result = execute_response_grpc
                     .result
-                    .expect("The action result is not defined.");
+                    .with_context(|| "The action result is not defined.")?;
 
                 let execution_metadata = action_result
                     .execution_metadata
-                    .expect("The execution metadata are not defined.");
+                    .with_context(|| "The execution metadata are not defined.")?;
+
+                let output_files: anyhow::Result<Vec<TFile>> = action_result
+                    .output_files
+                    .into_iter()
+                    .map(|output_file| {
+                        let output_file_digest =
+                            output_file.digest.with_context(|| "Digest not found.")?;
+
+                        Ok(TFile {
+                            digest: DigestWithStatus {
+                                status: tstatus_ok(),
+                                digest: tdigest_from(output_file_digest),
+                                _dot_dot_default: (),
+                            },
+                            name: output_file.path,
+                            existed: false,
+                            executable: output_file.is_executable,
+                            ttl: 0,
+                            _dot_dot_default: (),
+                        })
+                    })
+                    .collect();
+
+                let output_directories: anyhow::Result<Vec<TDirectory2>> = action_result
+                    .output_directories
+                    .into_iter()
+                    .map(|output_directory| {
+                        let digest = tdigest_from(
+                            output_directory
+                                .tree_digest
+                                .with_context(|| "Tree digest not defined.")?,
+                        );
+                        Ok(TDirectory2 {
+                            path: output_directory.path,
+                            tree_digest: digest.clone(),
+                            root_directory_digest: digest,
+                            _dot_dot_default: (),
+                        })
+                    })
+                    .collect();
```

Comment:
Usually you would deal with the error immediately - so `.collect::<anyhow::Result<Vec<_>>>()?` instead of having an anyhow float around for a while.